server/container/guice/protocols/imap/src/main/java/org/apache/james/modules/protocols/IMAPServerModule.java

```diff
@@ -99,6 +100,8 @@ protected void configure() {
         bind(MailboxTyper.class).to(DefaultMailboxTyper.class).in(Scopes.SINGLETON);
 
         Multibinder.newSetBinder(binder(), GuiceProbe.class).addBinding().to(ImapGuiceProbe.class);
+
+        Multibinder.newSetBinder(binder(), AbstractServerFactory.class).addBinding().to(IMAPServerFactory.class);
```

Comment:
could we declare a interface for that instead of an abstract class (that breaks the SRP)?

Reply:
OK